src/include/concurrency/transaction_manager.h

```diff
@@ -172,11 +183,14 @@ class TransactionManager {
   // first value is exclusive, last value is inclusive
   std::pair<cid_t, cid_t> dirty_range_ =
       std::make_pair(INVALID_CID, INVALID_CID);
+  
+  bool is_from_transaction_pool_;
```

Comment:
Is this just to make sure asserts fail in the correct cases? If so, it should be clear that it is slated for removal.